apps/roam/src/utils/settings.ts

```diff
@@ -0,0 +1,34 @@
+import getExtensionAPI from "roamjs-components/util/extensionApiContext";
+import localStorageGet from "roamjs-components/util/localStorageGet";
+
+export function getSetting<T>(key: string, defaultValue?: T): T {
+  const extensionAPI = getExtensionAPI();
+  const value = extensionAPI.settings.get(key);
+
+  if (value !== undefined && value !== null) {
+    return value as T;
+  }
+
+  // Fall back to localStorage for backward compatibility then migrate to extension settings
+  const roamjsKey = `roamjs:${key}`;
+  const roamjsKey2 = `roamjs:${key}:${window.roamAlphaAPI.graph.name}`;
+  const localValue =
+    localStorageGet(key) ||
+    localStorageGet(roamjsKey) ||
+    localStorageGet(roamjsKey2);
+
+  if (localValue !== null && localValue !== undefined) {
+    extensionAPI.settings.set(key, localValue);
+    localStorage.removeItem(key);
+    localStorage.removeItem(roamjsKey);
+    localStorage.removeItem(roamjsKey2);
+    return localValue as T;
+  }
+
+  return defaultValue as T;
+}
```

Comment:
Nice idea, but might be overkill.  GitHub Sync shouldn't need it, because it is a new feature.  

Do you already envision where/when we might use this?

Reply:
oh nice. yeah i forgot that's a new feature. i was just looking at the application storage to figure out which one needs migration. maybe we can just remove it? it doesn't seem like we use localstorage for any feature outside of github sync for now

Reply:
Yeah, let's remove it for now and bring it back if/when needed. 👍